quota/etcd/README.md

```diff
@@ -0,0 +1,246 @@
+1.1) is recommended, so there is some protection against token leakage without
+too much compromise of the quota system on exceptional situations.
+
+### Monitoring
+
+The following metrics are relevant when considering quota behavior:
+
+* [interceptor_request_count](https://github.com/google/trillian/blob/3cf59cdfd0/server/interceptor/interceptor.go#L91)
+* [interceptor_request_denied_count](https://github.com/google/trillian/blob/3cf59cdfd0/server/interceptor/interceptor.go#L95)
+* [quota_acquired_tokens](https://github.com/google/trillian/blob/3cf59cdfd0/quota/metrics.go#L70)
+* [quota_returned_tokens](https://github.com/google/trillian/blob/3cf59cdfd0/quota/metrics.go#L71)
+* [quota_replenished_tokens](https://github.com/google/trillian/blob/3cf59cdfd0/quota/metrics.go#L71)
+
+Requests denied due to token shortage are labeled on
+**interceptor_request_denied_count** as
+[insufficient_tokens](https://github.com/google/trillian/blob/3cf59cdfd0/server/interceptor/interceptor.go#L38).
+The ratio between **denied_with_insufficient_tokens** and
+**interceptor_request_count** is a strong indicator of token outages.
+
+## General concepts
+
+Trillian quotas have a finite number of tokens that get consumed (subtracted) by
+requests. Once a quota reaches zero tokens, all requests that would otherwise
+consume a token from it will fail with a **resource_exhausted** error. Tokens
+are replenished (added) by different mechanisms, depending on the quota
+configuration (e.g, X tokens every Y seconds).
+
+Quotas are designed so that a set of quotas, in different levels of granularity,
+apply to a single request.
+
+A quota
+[Spec](https://github.com/google/trillian/blob/3cf59cdfd0/quota/quota.go#L56)
+identifies a particular quota and represents to which requests its token count
+applies to. Specs contain a
+[Group](https://github.com/google/trillian/blob/3cf59cdfd0/quota/quota.go#L27)
+(global, tree and user) and
+[Kind](https://github.com/google/trillian/blob/3cf59cdfd0/quota/quota.go#L44)
+(read or write).
+
+A few Spec examples are:
+
+* global/read (all read requests)
+* global/write (all write requests)
+* trees/123/write (write requests for tree 123)
+* users/alice/read (read requests made by user "alice")
+
+Each request, depending on whether it's a read or write request, subtracts
+tokens from the following Specs:
+
+* users/$id/read or users/$id/write
+* trees/$id/read or trees/$id/write
+* global/read or global/write
+
+Quotas that aren't explicitly configured are considered infinite, thus they'll
+never block requests.
+
+## Etcd quotas
+
+Etcd quotas implement the concepts described above by storing the quota
+configuration and token count in etcd.
+
+Two replenishment mechanisms are available: sequencing-based and time-based.
+
+Sequencing-based replenishment is tied to logsigner's progress. A token is
+restored for each leaf sequenced from the Unsequenced table. As such, it's only
+applicable to global/write and tree/write quotas.
```

Comment:
`global/...`

Reply:
Done throughout, except in the new table, as I found it unnecessary there (and a bit ugly tbh).